src/git/zcl_abapgit_git_porcelain.clas.abap

```diff
@@ -4,31 +4,32 @@ CLASS zcl_abapgit_git_porcelain DEFINITION
   CREATE PUBLIC .
 
   PUBLIC SECTION.
-
-    TYPES:
-      BEGIN OF ty_pull_result,
-        files   TYPE zif_abapgit_definitions=>ty_files_tt,
-        objects TYPE zif_abapgit_definitions=>ty_objects_tt,
-        branch  TYPE zif_abapgit_definitions=>ty_sha1,
-      END OF ty_pull_result .
-    TYPES:
-      BEGIN OF ty_push_result,
-        new_files     TYPE zif_abapgit_definitions=>ty_files_tt,
-        branch        TYPE zif_abapgit_definitions=>ty_sha1,
-        updated_files TYPE zif_abapgit_definitions=>ty_file_signatures_tt,
-        new_objects   TYPE zif_abapgit_definitions=>ty_objects_tt,
-      END OF ty_push_result .
+    TYPES: BEGIN OF ty_pull_result,
+             files   TYPE zif_abapgit_definitions=>ty_files_tt,
+             objects TYPE zif_abapgit_definitions=>ty_objects_tt,
+             branch  TYPE zif_abapgit_definitions=>ty_sha1,
+             commit  TYPE zif_abapgit_definitions=>ty_sha1,
```

Comment:
the branch and commit both is a commit sha1, so both are not needed at the same time?

Reply:
Yes, but we need it to separate a branch-related commit from a "normal" one:
![grafik](https://user-images.githubusercontent.com/45243391/74030679-e9de8400-49af-11ea-847a-fa576460e95d.png)